source/benchmarking/benchmarking.md

```diff
@@ -0,0 +1,689 @@
+its utter simplicity given that it needs to be implemented identically across multiple languages for every driver.*
+
+The 50th percentile (i.e. the median) will be used for score composition. Other percentiles will be stored for
+visualizations and analysis (e.g. a "candlestick" chart showing benchmark volatility over time).
+
+Each task will have defined for it an associated size in megabytes (MB). The score for micro-benchmark composition will
+be the task size in MB divided by the median wall clock time.
+
+## Micro-benchmark definitions
+
+Datasets are available in the `data` directory adjacent to this spec.
+
+Note: The term "LDJSON" means "line-delimited JSON", which should be understood to mean a collection of UTF-8 encoded
+JSON documents (without embedded CR or LF characters), separated by a single LF character. (Some Internet definition of
+line-delimited JSON use CRLF delimiters, but this benchmark uses only LF.)
+
+### BSON micro-benchmarks
+
+Datasets are in the 'extended_bson' tarball.
+
+BSON tests focus on BSON encoding and decoding; they are client-side only and do not involve any transmission of data to
+or from the benchmark server. When appropriate, data sets will be stored on disk as
+[extended strict JSON](https://www.mongodb.com/docs/manual/reference/mongodb-extended-json). For drivers that don't
+support extended JSON, a BSON analogue will be provided as well.
+
+BSON micro-benchmarks include:
+
+- Flat BSON Encoding and Flat BSON Decoding -- shallow documents with only common BSON field types
+- Deep BSON Encoding and Deep BSON Decoding -- deeply nested documents with only common BSON field types
+- Full BSON Encoding and Full BSON Decoding -- shallow documents with all possible BSON field types
+
+#### Flat BSON Encoding
+
+Summary: This benchmark tests driver performance encoding documents with top level key/value pairs involving the most
+commonly-used BSON types.
+
+Dataset: The dataset, designated FLAT_BSON (ftnt4 Disk file 'flat_bson.json'), will be synthetically generated and
+consist of an extended JSON document with a single `_id` key with an object ID value plus 24 top level keys/value pairs
+of the following types: string, Int32, Int64, Double, Boolean. (121 total key/value pairs) Keys will be random ASCII
+strings of length 8. String data will be random ASCII strings of length 80.
+
+Dataset size: For score purposes, the dataset size for a task is the size of the single-document source file (7531
+bytes) times 10,000 operations, which equals 75,310,000 bytes or 75.31 MB.
+
+Phases:
+
+|             |                                                                                                                                                                                              |
+| ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```

Comment:
This and all other tables here have an empty row at the top.

Reply:
Github Flavored Markdown doesn't support tables without headers

Reply:
I'll make up some headers.  :smile:

Reply:
Oh, I wasn't aware of that. Thanks for making up some headers!